addons/website_slides/tests/test_karma.py

```diff
@@ -54,7 +54,14 @@ def test_karma_gain(self):
         # Finish the Course
         self.slide.with_user(user).action_set_completed()
         self.assertFalse(self.channel.with_user(user).completed)
-        (self.slide_2 | self.slide_3).with_user(user).action_set_completed()
+        self.slide_2.with_user(user).action_set_completed()
+
+        # answer a quizz question
+        self.slide_3.with_user(user).action_set_viewed(quiz_attempts_inc=True)
+        self.slide_3.with_user(user)._action_set_quiz_done()
+        self.slide_3.with_user(user).action_set_completed()
+        computed_karma += self.slide_3.quiz_first_attempt_reward
```

Comment:
Could add a `self.assertEqual(user.karma, computed_karma)` after this line.

Otherwise it's very nice sir!

Reply:
actually no, the test fails otherwise because both `quiz_first_attempt_reward` and `karma_gen_channel_finish` are added at the same time when confirming the last slide

Reply:
Oops, my bad then